java/test/src/main/java/io/ray/test/MetricTest.java

```diff
@@ -84,4 +134,129 @@ public void testAddHistogram() {
+    TestUtils.skipTestUnderSingleProcess();
+    Sum sum = registerSum();
+
+    sum.update(10.0);
+    sum.update(20.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 30.0));
+    TimeUnit.MILLISECONDS.sleep(MetricConfig.DEFAULT_CONFIG.timeIntervalMs() + 1000L);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 0.0));
+    sum.update(1.0);
+    sum.update(2.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 3.0));
+  }
+
+  @Test
+  public void testRegisterHistogram() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    Histogram histogram = registerHistogram();
+
+    for (int i = 1; i <= 200; ++i) {
+      histogram.update(i * 1.0d);
+    }
+    Assert.assertTrue(doubleEqual(histogram.getValue(), 200.0));
+    List<Double> window = histogram.getHistogramWindow();
+    for (int i = 0; i < Histogram.HISTOGRAM_WINDOW_SIZE; ++i) {
+      Assert.assertTrue(doubleEqual(i + 101.0d, window.get(i)));
+    }
+    TimeUnit.MILLISECONDS.sleep(MetricConfig.DEFAULT_CONFIG.timeIntervalMs() + 1000L);
+    Assert.assertTrue(doubleEqual(histogram.getValue(), 0.0));
+  }
+
+  @Test
+  public void testRegisterGaugeWithConfig() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    MetricConfig config = initRayMetrics(3000L, 1, 1000L);
+    Gauge gauge = registerGauge();
+
+    gauge.update(2.0);
+    Assert.assertTrue(doubleEqual(gauge.getValue(), 2.0));
+    TimeUnit.MILLISECONDS.sleep(config.timeIntervalMs() + 1000L);
+    Assert.assertTrue(doubleEqual(gauge.getValue(), 0.0));
+    gauge.update(5.0);
+    Assert.assertTrue(doubleEqual(gauge.getValue(), 5.0));
+  }
+
+  @Test
+  public void testRegisterCountWithConfig() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    MetricConfig config = initRayMetrics(3000L, 1, 1000L);
+    Count count = registerCount();
+
+    count.inc(10.0);
+    count.inc(20.0);
+    Assert.assertTrue(doubleEqual(count.getCount(), 30.0));
+    TimeUnit.MILLISECONDS.sleep(config.timeIntervalMs() + 1000L);
+    Assert.assertTrue(doubleEqual(count.getCount(), 0.0));
+    count.inc(1.0);
+    count.inc(2.0);
+    Assert.assertTrue(doubleEqual(count.getCount(), 3.0));
+  }
+
+  @Test
+  public void testRegisterSumWithConfig() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    MetricConfig config = initRayMetrics(3000L, 1, 1000L);
+    Sum sum = registerSum();
+
+    sum.update(10.0);
+    sum.update(20.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 30.0));
+    TimeUnit.MILLISECONDS.sleep(config.timeIntervalMs() + 1000L);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 0.0));
+    sum.update(1.0);
+    sum.update(2.0);
+    Assert.assertTrue(doubleEqual(sum.getSum(), 3.0));
+  }
+
+  @Test
+  public void testRegisterHistogramWithConfig() throws InterruptedException {
+    TestUtils.skipTestUnderSingleProcess();
+    MetricConfig config = initRayMetrics(3000L, 1, 1000L);
```

Comment:
Do not sleep in test to avoid time-consuming in ut.

Reply:
Fixed